File: C++ solutions/283. Move Zeroes.cpp
//approach : 1
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int k=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){  
                nums[k++]=nums[i];
            }
        }
        for(int i=k;i<nums.size();i++) nums[i]=0;

    }
};

//approach : 2
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n= nums.size();
        int left=0, end=n-1,right=0;
        while(right<=end){
            if(nums[right]!=0){
                swap(nums[left],nums[right]);                  
                left++;
            }
            right++;
        }
    }
};

File: JavaScript solutions/283. Move Zeroes.js
//approach : 1
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let k = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] != 0) {
      nums[k++] = nums[i];
    }
  }
  for (let i = k; i < nums.length; i++) nums[i] = 0;
};

//approach : 2
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let n = nums.length;
  let left = 0,
    end = n - 1,
    right = 0;
  while (right <= end) {
    if (nums[right] != 0) {
      [nums[left], nums[right]] = [nums[right], nums[left]];
      left++;
    }
    right++;
  }
};
